rand_distr/tests/pdf.rs

```diff
@@ -74,13 +82,101 @@ fn normal() {
         // Difference larger than 3 standard deviations or cutoff
         let tol = (3. * e).max(1e-4);
         if d > tol {
             panic!("Difference = {} * tol", d / tol);
         }
     }
-}
\ No newline at end of file
+}
+
+#[test]
+fn skew_normal() {
+    const N_SAMPLES: u64 = 1_000_000;
+    const LOCATION: f64 = 2.;
+    const SCALE: f64 = 0.5;
+    const SHAPE: f64 = -3.0;
+    const MIN_X: f64 = -1.;
+    const MAX_X: f64 = 4.;
+
+    let dist = SkewNormal::new(LOCATION, SCALE, SHAPE).unwrap();
+    let mut hist = Histogram100::with_const_width(MIN_X, MAX_X);
+    let mut rng = rand::rngs::SmallRng::seed_from_u64(1);
+
+    for _ in 0..N_SAMPLES {
+        let _ = hist.add(rng.sample(dist)); // Ignore out-of-range values
+    }
+
+    println!(
+        "Sampled skew normal distribution:\n{}",
+        sparkline::render_u64_as_string(hist.bins())
+    );
+
+    fn pdf(x: f64) -> f64 {
+        let x_normalized = (x - LOCATION) / SCALE;
+        let normal_density_x =
+            (-0.5 * (x_normalized).powi(2)).exp() / (2. * core::f64::consts::PI).sqrt();
+        let normal_distribution_x = 0.5
+            * (1.0 + statrs::function::erf::erf(SHAPE * x_normalized / core::f64::consts::SQRT_2));
+        2.0 / SCALE * normal_density_x * normal_distribution_x
+    }
+
+    let mut bin_centers = hist.centers();
+    let mut expected = [0.; HIST_LEN];
+    for e in &mut expected[..] {
+        *e = pdf(bin_centers.next().unwrap());
+    }
+
+    println!(
+        "Expected skew normal distribution:\n{}",
+        sparkline::render_u64_as_string(hist.bins())
+    );
+
+    let mut diff = [0.; HIST_LEN];
+    for (i, n) in hist.normalized_bins().enumerate() {
+        let bin = (n as f64) / (N_SAMPLES as f64);
+        diff[i] = (bin - expected[i]).abs();
+    }
+
+    println!(
+        "Difference:\n{}",
+        sparkline::render_f64_as_string(&diff[..])
+    );
+    println!(
+        "max diff: {:?}",
+        diff.iter().fold(core::f64::NEG_INFINITY, |a, &b| a.max(b))
+    );
+
+    // Check that the differences are significantly smaller than the expected error.
+    let mut expected_error = [0.; HIST_LEN];
+    // Calculate error from histogram
+    for (err, var) in expected_error.iter_mut().zip(hist.variances()) {
+        *err = var.sqrt() / (N_SAMPLES as f64);
+    }
+    // Normalize error by bin width
+    for (err, width) in expected_error.iter_mut().zip(hist.widths()) {
+        *err /= width;
+    }
+    // TODO: Calculate error from distribution cutoff / normalization
+
+    println!(
+        "max expected_error: {:?}",
+        expected_error
+            .iter()
+            .fold(core::f64::NEG_INFINITY, |a, &b| a.max(b))
+    );
+    for (&d, &e) in diff.iter().zip(expected_error.iter()) {
+        // Difference larger than 3 standard deviations or cutoff
+        let tol = (3. * e).max(1e-4);
+        if d > tol {
+            panic!("Difference = {} * tol", d / tol);
+        }
+    }
+}
```

Comment:
We should probably figure out how to deduplicate the code here, but this can be done in a follow-up PR.

Reply:
Yes! Although I am not so sure how to do the code fully generic to test discrete distributions in the future.

For now, for these continuous distributions, I would put together the repetitive code in a function that takes the variables `n_samples`, `min_x`, `max_x`, `dist`, `pdf` and `name` (name of the distribution being tested for reporting it). 

It would look like this:
```rust
#[test]
fn skew_normal() {
    const N_SAMPLES: u64 = 1_000_000;
    const LOCATION: f64 = 2.;
    const SCALE: f64 = 0.5;
    const SHAPE: f64 = -3.0;
    const MIN_X: f64 = -1.;
    const MAX_X: f64 = 4.;

    let dist = SkewNormal::new(LOCATION, SCALE, SHAPE).unwrap();
    fn pdf(x: f64) -> f64 { 
        // some code
    }
    histogram_test(N_SAMPLES, MIN_X, MAX_X, dist, pdf, "skew_normal");
}
```
where `histogram_test` takes the code repeated in both cases.

You see that the signature of `histogram_test` is not so clear for `dist` when testing discrete variables. And one would have to take into account the width of the bins of the histogram in that case too.